ui/page/components/consensus_list.go

```diff
@@ -0,0 +1,162 @@
+package components
+
+import (
+	"image/color"
+
+	"gioui.org/layout"
+	"gioui.org/text"
+
+	"github.com/planetdecred/dcrlibwallet"
+	"github.com/planetdecred/godcr/ui/decredmaterial"
+	"github.com/planetdecred/godcr/ui/load"
+	"github.com/planetdecred/godcr/ui/values"
+)
+
+var canVote bool
+
+type ConsensusItem struct {
+	Agenda     dcrlibwallet.Agenda
+	VoteButton decredmaterial.Button
+}
+
+func AgendasList(gtx C, l *load.Load, consensusItem *ConsensusItem) D {
+	gtx.Constraints.Min.X = gtx.Constraints.Max.X
+	return layout.UniformInset(values.MarginPadding16).Layout(gtx, func(gtx C) D {
+		agenda := consensusItem.Agenda
+		return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+			layout.Rigid(func(gtx C) D {
+				return layoutAgendaStatus(gtx, l, consensusItem.Agenda)
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layoutAgendaDescription(gtx, l, agenda)
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layoutAgendaID(gtx, l, agenda)
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layoutAgendaVotingPreference(gtx, l, agenda)
+			}),
+			layout.Rigid(func(gtx C) D {
+				return layoutAgendaVoteAction(gtx, l, consensusItem)
+			}),
```

Comment:
`layoutAgendaStatus`, `layoutAgendaDescription`, `layoutAgendaID`, `layoutAgendaVotingPreference`, `layoutAgendaVoteAction` can all fit into one method and the value is passed to it. Reason is they all use same codes.

```
func layoutAgendaDescription(gtx C, l *load.Load, agenda dcrlibwallet.Agenda) D {
	lbl := l.Theme.H6(agenda.Description)
	lbl.Font.Weight = text.Light
	return layout.Inset{Top: values.MarginPadding4}.Layout(gtx, lbl.Layout)
}

func layoutAgendaID(gtx C, l *load.Load, agenda dcrlibwallet.Agenda) D {
	lbl := l.Theme.H6("ID: #" + agenda.AgendaID)
	lbl.Font.Weight = text.Light
	return layout.Inset{Top: values.MarginPadding4}.Layout(gtx, lbl.Layout)
}

func layoutAgendaVotingPreference(gtx C, l *load.Load, agenda dcrlibwallet.Agenda) D {
	lbl := l.Theme.H6("Voting Preference: " + agenda.VotingPreference)
	lbl.Font.Weight = text.Light
	return layout.Inset{Top: values.MarginPadding4}.Layout(gtx, lbl.Layout)
}
```

Reply:
@Sirmorrison the current implementation enhances code readability